### PR TITLE
[RHOAIENG-2905] Pipeline Run Details tabs without content should be disabled

### PR DESCRIPTION
**Pull request by jpuzz0**
Closes: [RHOAIENG-2905](https://issues.redhat.com/browse/RHOAIENG-2905)

## Description
All tabs associated with pipeline details that had conditions that led to tab content containing "no content" text are now disabled instead.

e.g.
image

## Test Impact
Updated cypress test

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by manaswinidas — APPROVED** [review]
/lgtm Works fine.

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx [hunk 1]
```diff
@@ -35,35 +24,47 @@ const PipelineRunDrawerRightTabs: React.FC<PipelineRunDrawerRightTabsProps> = ({
   task,
   executions,
 }) => {
-  const [selection, setSelection] = React.useState(PipelineRunNodeTabs.INPUT_OUTPUT);
+  const [selection, setSelection] = React.useState<string>(PipelineRunNodeTab.INPUT_OUTPUT);
```
Comment (DaoDaoNoCode) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx:27]: What if `PipelineRunNodeTab.INPUT_OUTPUT` tab is disabled here and it's set as the default selection tab?

Comment (jpuzz0) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx]: Added a fallback to default to the "Details" tab when no inputs/outputs exist.

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx [hunk 2]
```diff
@@ -35,47 +24,64 @@ const PipelineRunDrawerRightTabs: React.FC<PipelineRunDrawerRightTabsProps> = ({
...
-    [PipelineRunNodeTabs.DETAILS]: <SelectedNodeDetailsTab task={task} />,
-    [PipelineRunNodeTabs.VOLUMES]: <SelectedNodeVolumeMountsTab task={task} />,
-    [PipelineRunNodeTabs.LOGS]: <LogsTab task={task} />,
-    // [PipelineRunNodeTabs.POD]: <>TBD 6</>,
-    // [PipelineRunNodeTabs.EVENTS]: <>TBD 7</>,
-    // [PipelineRunNodeTabs.ML_METADATA]: <>TBD 8</>,
+  const tabs: Record<string, { title: string; isDisabled?: boolean; content: React.ReactNode }> = {
+    [PipelineRunNodeTab.INPUT_OUTPUT]: {
+      title: 'Input/Output',
+      isDisabled: hasNoInputsOutputs,
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx:38]: do you need to put hasNoInputsOutputs as a variable, can it just be inline

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx [hunk 3]
```diff
@@ -35,47 +24,64 @@ const PipelineRunDrawerRightTabs: React.FC<PipelineRunDrawerRightTabsProps> = ({
...
       <DrawerPanelBody className="pipeline-run__drawer-panel-body pf-v5-u-px-sm">
-        <TabContent
-          id={selection}
-          eventKey={selection}
-          activeKey={selection}
-          style={{ flex: '1 1 auto' }}
-        >
-          {tabContents[selection]}
-        </TabContent>
+        {!tabs[selection].isDisabled && (
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx:75]: is this needed? `!tabs[selection].isDisabled`

Comment (jpuzz0) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx:75]: I wasn't sure if there was a scenario where maybe some isDisabled condition still leading to one of the default tabs (Summary or Inputs/Outputs) being displayed. isDisabled doesn't prevent the rendering of the content within the tab. If you want me to take it out considering this, let me know.

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx (hunk 2 above)
Comment (jpuzz0) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx:38]: Its used in the condition for the default tab state as well, which is why I made a variable for it. I can repeat the condition twice if you want.

Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx:38]: no i just thought i didnt see it used twice, but im seeing it now so nvm

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx (hunk 3 above)
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx:75]: doesnt really affect the logic, we can keep it

**Comment by Gkrumbach07** [comment]
cc @yannnz This was from an old jira to make these disabled, if you feel there should be another UX for no content let us know

/lgtm
/approve

**Pull request merged**